internal/ls/utilities.go

```diff
@@ -1628,3 +1628,10 @@ func findContainingList(node *ast.Node, file *ast.SourceFile) *ast.NodeList {
 	astnav.VisitEachChildAndJSDoc(node.Parent, file, nodeVisitor)
 	return list
 }
+
+func getLeadingCommentRangesOfNode(node *ast.Node, file *ast.SourceFile) iter.Seq[ast.CommentRange] {
+	if node.Kind == ast.KindJsxText {
+		return nil
+	}
+	return scanner.GetLeadingCommentRanges(&ast.NodeFactory{}, file.Text(), node.Pos())
```

Comment:
It's so weird that this needs a factory because we only ever really need to look at a single comment range at a time.